src/Hedgehog/Property.fs

```diff
@@ -128,76 +136,80 @@ module private Pretty =
     let private append (sb : StringBuilder) (msg : string) : unit =
         sb.AppendLine msg |> ignore
 
-    let renderOK (tests : int<tests>) : string =
-        sprintf "+++ OK, passed %s." (renderTests tests)
+    let private renderf (sb : StringBuilder) (fmt : Printf.StringFormat<'a, unit>) : 'a =
+        Printf.ksprintf (sb.AppendLine >> ignore) fmt
+
+    let renderOK (report : Report) : string =
+        sprintf "+++ OK, passed %s." (renderTests report.Tests)
 
-    let renderGaveUp (tests : int<tests>) (discards : int<discards>) : string =
+    let renderGaveUp (report : Report) : string =
         sprintf "*** Gave up after %s, passed %s."
-            (renderDiscards discards)
-            (renderTests tests)
-
-    let renderFailed
-            (tests : int<tests>)
-            (discards : int<discards>)
-            (shrinks : int<shrinks>)
-            (journal : Journal) : string =
+            (renderDiscards report.Discards)
+            (renderTests report.Tests)
+
+    let renderFailed (failure : FailureReport) (report : Report) : string =
         let sb = StringBuilder ()
 
-        sprintf "*** Failed! Falsifiable (after %s%s%s):"
-            (renderTests tests)
-            (renderAndShrinks shrinks)
-            (renderAndDiscards discards)
-            |> append sb
+        renderf sb "*** Failed! Falsifiable (after %s%s%s):"
+            (renderTests report.Tests)
+            (renderAndShrinks failure.Shrinks)
+            (renderAndDiscards report.Discards)
+
+        List.iter (append sb) (Journal.toList failure.Journal)
 
-        List.iter (append sb) (Journal.toList journal)
+        if failure.RenderRecheck then
+            renderf sb "This failure can be reproduced by running:"
+            renderf sb "> Property.recheck (%d : Size) (%0A) <property>"
```

Comment:
I think this won't work in F# interactive, it'll render the record values without semicolon `;` e.g.

```f#
Property.recheck (2 : Size) ({ Value = 6795855140008116271UL Gamma = 17914027628548019555UL }) <property>
```

but what we'd want is

```f#
Property.recheck (2 : Size) ({ Value = 6795855140008116271UL; Gamma = 17914027628548019555UL }) <property>
```

```suggestion
            renderf sb "> Property.recheck (%d : Size) ({ Value = %A; Gamma = %A }) <property>"
```